src/sourceMachine.ts

```diff
@@ -171,13 +169,14 @@ export const makeSourceMachine = (params: {
 
   return createMachine<typeof sourceModel>(
     {
-      initial: 'checking_if_on_legacy_url',
+      initial: 'checking_initial_data',
       preserveActionOrder: true,
       context: {
         ...sourceModel.initialContext,
-        sourceRawContent: params.data?.text || null,
-        sourceID: params.data?.id || null,
-        sourceProvider: params.data ? 'registry' : null,
+        sourceRawContent: params.sourceRegistryData?.text || null,
+        sourceID: params.sourceRegistryData?.id || null,
+        sourceProvider: params.sourceRegistryData ? 'registry' : null,
+        sourceRegistryData: params.sourceRegistryData,
```

Comment:
this is the reason why I have removed `GetSourceFileSSR` - I have wanted to add this to the context right away so the `<VizHead />` could always consume this from here. But since that SSR-only query had a different "return type" it was creating a type mismatch and I didn't want to fight with this - so I've just reused the other query on the backend to achieve type symbiosis.